src/Spectre.libGaussianMixtureModelling/ExpectationRunnerOpt.h

```diff
@@ -73,13 +73,19 @@ class ExpectationRunnerOpt
         const unsigned numberOfComponents = (unsigned)components.size();
         for (unsigned k = 0; k < numberOfComponents; k++)
         {
+            // Using Kahan summation algorithm for improved accuracy
```

Comment:
Can we extract this into a reusable element?

Reply:
I've considered doing that, but in another PR.

Reply:
Add a task then.